docs/API-Reference/bpmn-js/importXML.md

```diff
@@ -10,3 +10,62 @@ grand_parent: "API Reference"
 ---
 
 # importXML
+
+`importXML` API parses and renders a BPMN 2.0 diagram.
+
+## Usage
+
+importXML(xml: _[String](#)_, bpmnDiagram: _[ModdleElement](#)_ or _[String](#)_): Promise<_[ImportXMLResult](#importxmlresult)_, _[ImportXMLError](#importxmlerror)_>
+
+* `xml`: The BPMN 2.0 XML.
+* `bpmnDiagram`: The BPMN diagram or the id of the diagram to render _(if not provided, the first one will be rendered)_.
+
+### Code Example
+
+```javascript
+const modeler = new BpmnJS();
+
+try {
+
+  const result = await modeler.importXML(xml);
+
+  const { warnings } = result;
+
+  console.log('Import warnings: ', warnings);
+} catch (err) {
+
+  console.log('Error happened: ', err.message, err.warnings);
+}
+```
+
+### Life-Cycle Events
+
+During the import, these [events](/events) will be fired:
+
+* [import.parse.start](/events/bpmn-events/import-parse-start): About to read model from xml
+* [import.parse.complete](/events/bpmn-events/import-parse-complete): Model is read, either successfully or with errors.
+* [import.render.start](/events/bpmn-events/import-render-start): Graphical import is starting
+* [import.render.complete](/events/bpmn-events/import-render-complete): Graphical import is completed
+* [import.done](/events/bpmn-events/import-done): Everything is done.
+
+## Using with Callback Function
+
+With the [v7.0.0 release](https://github.com/bpmn-io/bpmn-js/blob/develop/CHANGELOG.md#700) public bpmn-js APIs are made awaitable and passing callbacks to APIs will be deprecated in a future release. See [here](https://bpmn.io/l/moving-to-promises.html) for more info.
+
+# ImportXMLResult
```

Comment:
Shouldn't this be in another `.md`? OR in a sub-section of `# importXML` named `## Returns` or something like that?

Reply:
I added this here for these 2 reasons:

1) We want permalink functionality: When you click on the ImportXMLResult link in the input/output parameters section, the page would scroll down to this header.
2) This is not a shared type, we only use this pseudo-type for the importXML API. We use this same pseudo-type in our codebase while documenting the importXML API.

Reply:
Does this make sense?

Reply:
Yeah, makes sense. Ty